opendevin/server/agent/agent.py

```diff
@@ -182,14 +182,18 @@ async def start_task(self, start_event):
         if 'task' not in start_event['args']:
             await self.send_error('No task specified')
             return
-        await self.send_message('Starting new task...')
+        # if agent task is already running, send chat
+        if self.agent_task and self.controller:
+            await self.controller.add_user_message(
+                UserMessageObservation(start_event['args']['task'])
+            )
+            return
```

Comment:
If the agent task is currently running, we directly send an `UserMessageObservation` to the agent instead of killing the task and re-start. Each agent can choose to keep that behavior by implementing separate logic to handle `UserMessageObservation`.